File: PAT_Basic_Level_by_java/1063 计算谱半径/1063 计算谱半径.java
/**
 * 1063 计算谱半径
 * 
 * 题目描述：
 * 给定 N 个复特征值的实部和虚部，计算矩阵的谱半径。
 * 谱半径定义为所有特征值的模的最大值，其中模 = sqrt(实部^2 + 虚部^2)。
 * 结果四舍五入保留 2 位小数。
 * 
 * 实现原理：
 * 解题思路：
 * 1. 遍历每组实部 a 和虚部 b，计算模 sqrt(a² + b²)。
 * 2. 记录所有模中的最大值作为谱半径。
 * 3. 使用 System.out.printf 格式化输出，保留两位小数。
 * 
 * 算法核心逻辑：遍历计算模 → 记录最大值 → 格式化输出。
 * 
 * 时间复杂度：O(N)，空间复杂度：O(1)。
 */
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine().trim());

        double maxRadius = 0;
        // 遍历每组特征值，计算模并更新最大值
        for (int i = 0; i < N; i++) {
            String[] parts = br.readLine().trim().split("\\s+");
            int a = Integer.parseInt(parts[0]); // 实部
            int b = Integer.parseInt(parts[1]); // 虚部
            double radius = Math.sqrt(a * a + b * b); // 计算模
            if (radius > maxRadius) {
                maxRadius = radius; // 更新谱半径最大值
            }
        }

        // 四舍五入保留两位小数输出
        System.out.printf("%.2f\n", maxRadius);
    }
}

File: PAT_Basic_Level_by_c/1063 计算谱半径/1063 计算谱半径.c
/*
 * 题目：1063 计算谱半径
 * 实现原理：
 *   计算复数集合的谱半径（即模最大的复数的模）。
 *   复数a+bi的模 = sqrt(a^2 + b^2)
 *   算法步骤：
 *   1. 读取N个复数（每个复数由a和b组成）
 *   2. 计算每个复数的模
 *   3. 记录最大的模
 *   4. 输出最大模（保留两位小数）
 */
#include <stdio.h>
#include <math.h>

int main() {
    int N;
    scanf("%d", &N);
    
    double max_radius = 0;  // 记录模的最大值
    for (int i = 0; i < N; i++) {
        int a, b;  // 复数的实部与虚部
        scanf("%d %d", &a, &b);
        double radius = sqrt(a * a + b * b);  // 计算该复数的模
        if (radius > max_radius) {  // 更新最大模
            max_radius = radius;
        }
    }
    
    printf("%.2f\n", max_radius);  // 谱半径保留两位小数输出
    
    return 0;
}